src/draw/lv_draw.h

```diff
@@ -36,11 +35,74 @@ extern "C" {
 /**********************
  *      TYPEDEFS
  **********************/
+typedef struct _lv_draw_backend_t{
+    struct _lv_draw_backend_t * base;
+
+    void * ctx;
+
+    void (*draw_rect)(const lv_area_t * coords, const lv_area_t * clip, const lv_draw_rect_dsc_t * dsc);
+
+    void (*draw_arc)(lv_coord_t center_x, lv_coord_t center_y, uint16_t radius,  uint16_t start_angle, uint16_t end_angle,
+                     const lv_area_t * clip_area, const lv_draw_arc_dsc_t * dsc);
+
+    void (*draw_img)(const lv_area_t * map_area, const lv_area_t * clip_area,
```

Comment:
Please let implementation handle all the logic (The same as `lv_draw_img`). By doing so, for implementations uses texture, they can process `src` on their own.

Reply:
It'd skip LVGL's image decoder interface. E.g. if a user registers a special custom decoder, its result should be passed to the drawers.